packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -260,9 +265,9 @@ IFluidDataStoreChannel, IFluidDataStoreRuntime, IFluidHandleContext {
         }
 
         this.attachListener();
-        // If exists on storage or loaded from a snapshot, it should already be binded.
-        this.bindState = existing ? BindState.Bound : BindState.NotBound;
-        this._attachState = dataStoreContext.attachState;
+
+        this.localAttachState = existing ? AttachState.Attached : AttachState.Detached;
```

Comment:
This is important because currently remote data stores' graphAttachState is always Detached.